semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala

```diff
@@ -30,81 +34,41 @@ class Main(settings: Settings, reporter: Reporter) {
       if (settings.par) settings.classpath.entries.par
       else settings.classpath.entries
 
-    val buffer = new ConcurrentLinkedQueue[AbsolutePath]()
-
-    def createCachedJar(cacheEntry: AbsolutePath)(f: AbsolutePath => Boolean): Unit =
-      MetacpGlobalCache.computeIfAbsent(cacheEntry.toNIO) { tmp =>
-        PlatformFileIO.withJarFileSystem(AbsolutePath(tmp), create = true) { out =>
-          val res = f(out)
-          success.compareAndSet(true, res)
-          buffer.add(cacheEntry)
-          res
-        }
-      }
-
-    if (!Files.exists(Files.createDirectories(settings.cacheDir.toNIO))) {
-      Files.createDirectories(settings.cacheDir.toNIO)
-    }
-
     classpath.foreach { entry =>
       if (entry.isDirectory) {
-        val out = AbsolutePath(Files.createTempDirectory("semanticdb"))
-        val semanticdbIndex = new SemanticdbIndex
-        val res = convertClasspathEntry(entry, out, semanticdbIndex)
-        semanticdbIndex.save(out)
-        success.compareAndSet(true, res)
-        buffer.add(out)
+        success.compareAndSet(true, convertClasspathEntry(entry, outRoot))
```

Comment:
I'd prefer if we removed parallel processing from metacp as well. Since metacp's performance is no longer a feature, I think it'd be great to remove all complications that are there solely for performance.

Reply:
The most important user of --par is in fact ConvertSuite because it shaves off ~35s for `slow:test` on my machine (just tested right now). That's a significant productivity difference when iterating on changes locally. I vote to keep --par.

Reply:
As discussed, I'm fine with keeping --par for now, but we'll need to return to the original behavior of multiple inputs => multiple outputs.